packages/remix-dev/compiler/assets.ts

```diff
@@ -56,10 +56,13 @@ export async function createAssetsManifest(
     config.appDirectory,
     config.entryClientFile
   );
-  let routesByFile: Map<string, Route> = Object.keys(config.routes).reduce(
+  let routesByFile: Map<string, Route[]> = Object.keys(config.routes).reduce(
     (map, key) => {
       let route = config.routes[key];
-      map.set(route.file, route);
+      map.set(
+        route.file,
+        map.has(route.file) ? [...map.get(route.file), route] : [route]
+      );
```

Comment:
How strict is the requirement here to use a map?

By no longer requiring a 1-1 mapping, we're defeating the point of using a map in the first place. An object might be simpler, but ultimately, doesn't make much difference, so I'll leave it up to you.

so instead of:
```js
Object.keys(config.routes).reduce((map, key) => {
  const route = config.routes[key];
  map.set(
    route.file,
    map.has(route.file) ? [...map.get(route.file), route] : [route],
  );
  return map;
}, new Map());
```

we would have:
```js
Object.keys(config.routes).reduce((map, key) => {
  const route = config.routes[key];
  map[route.file] = map[route.file] ? [...map[route.file], route] : [route];
  return map;
}, {});
```

Reply:
Hi @runmoore 

Today implementation built by remix core team are also made with `new Map()`:

https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler/assets.ts#L65

So I keep it in this way to work on their pattern.

Reply:
you're right, I was getting confused by `Map()` and `Set()`